microdnf/utils.cpp

```diff
@@ -193,4 +197,114 @@ std::string detect_release(const std::string & install_root_path) {
     return "";
 }
 
+// transaction details columns
+enum { COL_NEVRA, COL_REPO, COL_SIZE };
+
+/*static gint
+dnf_package_cmp_cb (DnfPackage **pkg1, DnfPackage **pkg2)
+{
+  return dnf_package_cmp (*pkg1, *pkg2);
+}*/
+
+static void add_transaction_packages(struct libscols_table *tb, struct libscols_line *parent, std::vector<libdnf::rpm::Package> & pkgs) {
+  // sort packages by NEVRA
+  //g_ptr_array_sort (pkgs, (GCompareFunc) dnf_package_cmp_cb);
+
+//  HyGoal goal = dnf_context_get_goal (ctx);
+  for (auto & pkg : pkgs) {
+      struct libscols_line *ln = scols_table_new_line(tb, parent);
+      scols_line_set_data(ln, COL_NEVRA, pkg.get_full_nevra().c_str());
+      scols_line_set_data(ln, COL_REPO, pkg.get_repo()->get_id().c_str());
+      //g_autofree gchar *formatted_pkg_size = g_format_size (dnf_package_get_size (pkg));
+      //scols_line_set_data(ln, COL_SIZE, formatted_pkg_size);
+      scols_line_set_data(ln, COL_SIZE, std::to_string(pkg.get_size()).c_str());
+
+/*      if (dnf_package_get_action (pkg) != DNF_STATE_ACTION_REMOVE)
+        {
+          g_autoptr(GPtrArray) pkgs_replaced = hy_goal_list_obsoleted_by_package (goal, pkg);
+          g_ptr_array_sort (pkgs_replaced, (GCompareFunc) dnf_package_cmp_cb);
+          for (guint i = 0; i < pkgs_replaced->len; i++)
+            {
+              DnfPackage *pkg = pkgs_replaced->pdata[i];
+              struct libscols_line *replacing_ln = scols_table_new_line (tb, ln);
+              g_autofree gchar *replacing_text = g_strconcat ("replacing ", dnf_package_get_nevra (pkg), NULL);
+              scols_line_set_data (replacing_ln, COL_NEVRA, replacing_text);
+            }
+        }*/
```

Comment:
Is it here still for some purpose? If yes, please add todo.

Reply:
It was for me. What is missing and how it is done in old microdnf. I will remove the the comments and add TODO.